core/src/opcode.cpp

```diff
@@ -193,6 +195,64 @@ Opcode decode(const uint8_t op) {
     }
 }
 
+MemoryAccess get_memory_access(const Family family) {
+    switch (family) {
+    case Family::Invalid:
+    case Family::BRK:
+    case Family::PHP:
+    case Family::BPL:
+    case Family::CLC:
+    case Family::BIT:
+    case Family::PLP:
+    case Family::AND:
+    case Family::JSR:
+    case Family::BMI:
+    case Family::SEC:
+    case Family::LSR:
+    case Family::PHA:
+    case Family::JMP:
+    case Family::BVC:
+    case Family::CLI:
+    case Family::ADC:
+    case Family::PLA:
+    case Family::RTS:
+    case Family::BVS:
+    case Family::SEI:
+    case Family::TXS:
+    case Family::BCC:
+    case Family::LDX:
+    case Family::LDY:
+    case Family::LDA:
+    case Family::BCS:
+    case Family::CLV:
+    case Family::BNE:
+    case Family::CLD:
+    case Family::CPX:
+    case Family::NOP:
+    case Family::INX:
+    case Family::INY:
+    case Family::CPY:
+    case Family::CMP:
+    case Family::BEQ:
+    case Family::SED:
+    case Family::TYA:
+    case Family::TAY:
+    case Family::TAX:
+    case Family::TSX:
+    case Family::TXA:
+    case Family::DEY:
+    case Family::DEX:
+        return MemoryAccess::Read;
+    case Family::STY:
+    case Family::STA:
+    case Family::STX:
+        return MemoryAccess::Write;
+    case Family::INC:
+        return MemoryAccess::ReadWrite;
+    }
+    return MemoryAccess::Read;
```

Comment:
Do you need this? Getting here would be UB.

Reply:
Yeah, UB or an instruction was not added. Changed to exception instead.